src/Lib/2D/Surface.hpp

```diff
@@ -58,8 +58,8 @@ class Surface : public NoCopy {
   void setOffsetCenter();
 
   // Accessor functions.
-  unsigned int getWidth() const { return twidth; }
-  unsigned int getHeight() const { return theight; }
+  unsigned short getWidth() const { return twidth; }
```

Comment:
I'm not sure if this is the best way to fix this. These two are multiplied on L757 of Surface.cpp

`memset(mem, color, getWidth() * getHeight() * sizeof(PIX));` The third argument of memset can apparently be size_t

`void *memset(void s[.n], int c, size_t n);`

Reply:
@Jammyjamjamman

ChatGPT:

```
In general, the result of multiplying two unsigned short int values and a uint8_t (Uint8) value is less likely to overflow a size_t variable compared to larger integer types. However, it still depends on the specific values involved.

...

However, it's important to note that the actual size of unsigned short int can vary depending on the platform. If the unsigned short int is 16 bits, the maximum value is 65535, and overflow is less likely. If it's larger (e.g., 32 bits), the maximum value would be larger, and overflow could occur with different input values.
```

Casting is mentioned at https://github.com/netpanzer/netpanzer/security/code-scanning/30 but I feel like the value might not come out right if overflow was prevented by casting. I'm not experienced with using numbers in my life.

Reply:
Yeah, I don't think we'll be creating surfaces of greater than 65k pixels, since they are bound to the monitor size.

Reply:
> but I feel like the value might not come out right if overflow was prevented by casting

I think your feeling is right. Have a look at this example: https://cplayground.com/?p=alligator-shrew-ibis
In clang++ you multiply 2 unsigned shorts together without casting one of them to size_t, it will try to multiply the values and store them in a (signed) int, which overflows. Then it puts the overflowed value in a size_t.

Reply:
I would just cast the `sizeof(PIX)` to an int be honest... :)

this never overflowed because nobody has a monitor > 25k x 25k pixels.

Reply:
@winrid I think changing how they're declared will fix other warnings from where they're used. I changed getPitch() as well.

Thanks for the feedback @Jammyjamjamman

Reply:
Yeah I know - which is cool (unless it causes the overflows jammy is talking about).